force-app/main/default/classes/DMLOperations.cls

```diff
@@ -114,8 +132,12 @@ public with sharing class DMLOperations {
     public static void updateAccountFields(Id accountId, String newName, String newIndustry) {
         // Query the account
         // [SELECT Id FROM Account WHERE Id = :accountId]
+        Account updatedAccount = [SELECT Id FROM Account WHERE Id = :accountId];
+        updatedAccount.Name = newName;
+        updatedAccount.Industry = newIndustry;
 
         // Update the Account with the given Id with the new name and industry
+        update updatedAccount;
```

Comment:
Good work. See my comment above. Is the SOQL query needed here?

Reply:
Account updatedAccount = new Account(Id = accountId);